### ISSUE TITLE
Small changes to RHOSR quarkus app

### ISSUE THREAD

**Issue by bibryam** [redhat-developer/app-services-guides#322]
Here are a few suggestions to change in the app. I haven't tried them to see if they break the app or not. 
cc @carlesarnal 

- Provide a default value for ${OAUTH_REALM} such as ${OAUTH_REALM:rhoas}
- Change io.apicurio.registry.serde.avro.strategy.RecordIdStrategy to io.apicurio.registry.serde.strategy.TopicIdStrategy The latter is what the UI will display if the strategy is used. This changes should be in both producer and processor application.properties

**Comment by wtrocki** [comment]
> Provide a default value for ${OAUTH_REALM} such as ${OAUTH_REALM:rhoas}

We are providing full value of the Auth url in CLI/UI. When creating service account in CLI env file/json will have following string

 RHOAS_OAUTH_TOKEN_URL=https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token

**Comment by bibryam** [comment]
Are you saying I don't need to set OAUTH_REALM? If that is the case, it is fine. I want to hide it and not have instructions for it

Another change noticed:
Change OAUTH_SERVER_URL to  OAUTH_TOKEN_ENDPOINT_URI so that we can use a single env variables for both Kafka and SR?

**Comment by wtrocki** [comment]
It is strange that is required a combination of authurl + clientID and secret should be enough (this will not work with SBO as we do not have that value there). @carlesarnal do you know why we require this value - it seems that it is required but not sure what is being used.

+1 for OAUTH_TOKEN_ENDPOINT_URI

We going to swap both examples later in separate issues to values we have in env vars in our toolings
```
RHOAS_SERVICE_ACCOUNT_CLIENT_ID=srvc-a-fd-df7e-fdsf-b529-e6a24a557358
RHOAS_SERVICE_ACCOUNT_CLIENT_SECRET=asd-fsdf-4d1f-8963-6ab83de375aa
RHOAS_SERVICE_ACCOUNT_AUTH_TOKEN_URL=https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token
```

**Comment by carlesarnal** [comment]
> It is strange that is required a combination of authurl + clientID and secret should be enough (this will not work with SBO as we do not have that value there). @carlesarnal do you know why we require this value - it seems that it is required but not sure what is being used.
> 
> +1 for OAUTH_TOKEN_ENDPOINT_URI
> 
> We going to swap both examples later in separate issues to values we have in env vars in our toolings
> 
> ```
> RHOAS_SERVICE_ACCOUNT_CLIENT_ID=srvc-a-fd-df7e-fdsf-b529-e6a24a557358
> RHOAS_SERVICE_ACCOUNT_CLIENT_SECRET=asd-fsdf-4d1f-8963-6ab83de375aa
> RHOAS_SERVICE_ACCOUNT_AUTH_TOKEN_URL=https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token
> ```
> 
> We need more specific envs to avoid overriding existing ones (like AWS tokens etc.)

This is required here because right now the example is using Quarkus libraries which are using Registry serdes 2.0.1.Final and that version required this configuration. This has been fixed in 2.1.0.Final so once this is updated in Quarkus, we will be able to change it.

**Comment by carlesarnal** [comment]
> > It is strange that is required a combination of authurl + clientID and secret should be enough (this will not work with SBO as we do not have that value there). @carlesarnal do you know why we require this value - it seems that it is required but not sure what is being used.
> > +1 for OAUTH_TOKEN_ENDPOINT_URI
> > We going to swap both examples later in separate issues to values we have in env vars in our toolings
> > ```
> > RHOAS_SERVICE_ACCOUNT_CLIENT_ID=srvc-a-fd-df7e-fdsf-b529-e6a24a557358
> > RHOAS_SERVICE_ACCOUNT_CLIENT_SECRET=asd-fsdf-4d1f-8963-6ab83de375aa
> > RHOAS_SERVICE_ACCOUNT_AUTH_TOKEN_URL=https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > We need more specific envs to avoid overriding existing ones (like AWS tokens etc.)
> 
> This is required here because right now the example is using Quarkus libraries which are using Registry serdes 2.0.1.Final and that version required this configuration. This has been fixed in 2.1.0.Final so once this is updated in Quarkus, we will be able to change it.

I'll ping you here once I get the update inside a Quarkus release.

**Comment by carlesarnal** [comment]
This has been included in Quarkus 2.7.0.Final.

**Comment by wtrocki** [comment]
@carlesarnal Looking at this now. Example we use still uses some old version of quarkus:
https://github.com/redhat-developer/app-services-guides/blob/main/code-examples/quarkus-service-registry-quickstart/producer/pom.xml#L17

Would you know/can assist/create PR with required changes to get that sorted?
